src/coreclr/nativeaot/Runtime/arm64/WriteBarriers.S

```diff
@@ -116,12 +116,10 @@
 2:
         // We can skip the card table write if the reference is to
         // an object not on the epehemeral segment.
-        PREPARE_EXTERNAL_VAR_INDIRECT g_ephemeral_low, x12
-        cmp     \refReg, x12
-        blo     0f
-
-        PREPARE_EXTERNAL_VAR_INDIRECT g_ephemeral_high, x12
+        PREPARE_EXTERNAL_VAR_INDIRECT g_ephemeral_low,  x12
```

Comment:
I saw you have used `ldp` at one point. Is it not profitable here?

Reply:
I saw no visible benefits from it, while it's possible to use here for NativeAOT easily, it's a bit tricky on CoreCLR with its:
```asm
    ldr  x12, LOCAL_LABEL(wbs_ephemeral_low)
    ldr  x17, LOCAL_LABEL(wbs_ephemeral_high)
```
I wasn't sure how to properly use `ldp` with `LOCAL_LABEL` 🤔
I tried:
```
    ldp  x12, x17, LOCAL_LABEL(wbs_ephemeral_low) -- not compiling
    ldp  x12, x17, [sp, LOCAL_LABEL(wbs_ephemeral_low)] -- not compiling
```
I guess ldp expects some real sp delta rather than label imm.

From my understanding, the difference between CoreCLR and NativeAOT is that NativeAOT accesses GC variables directly, while on CoreCLR we have a mechanism that dumps them all to a "patchable pool" next to the Write-barrier function for better cache access so it can access them by label in code